### DISCUSSION TITLE
SimpleITK image registration

### DISCUSSION THREAD

**Discussion by Katitamorais** [IBEXImagingCommunity/ibex_imaging_knowledge_base#195] [General]
Hello!

We follow all instructions described in the IBEX Nature Protocols article for Image Registration. 

For some samples, we have a correlation close to 1 (melanoma sample), similar to the result observed with a IBEX training dataset (Kidney). However, for other samples correlation is around 0.8 (tonsil and leish).

In addition to the visual impact on the final aligned image, this also makes it difficult to individualize cells, define segmentation masks and quantification of phenotype.

Do you have any recommendation to improve our image registration?

Thanks.
[Image Registration Simple ITK.pptx](https://github.com/user-attachments/files/16382282/Image.Registration.Simple.ITK.pptx)

**Comment by zivy** [comment]
Hello @Katitamorais

Based solely on the correlation values after registration one cannot say if the registration is successful or not. A correlation value of 0.8 can indicate a registration that is just as good as a correlation value of 1 (obtaining perfect correlation is very rare). The question is, are corresponding structures aligned with each other? I assume they aren't, otherwise you wouldn't be here. Can you share an image showing the final results and the alignment discrepancies?

Some things to note:
1. The [XTRegisterSameChannel](https://niaid.github.io/imaris_extensions/XTRegisterSameChannel.html) program used in the original IBEX protocols paper uses an affine transformation (global translation+rotation+scale+shear). If the tissue undergoes local deformations the registration cannot align it accurately. It is a feature of the registration, not a bug. If your tissues undergo local deformations you could try the [elastix-napari plugin](https://www.napari-hub.org/plugins/elastix-napari) which also supports a BSpline Transform (cannot vouch for it as I haven't used it, but it's out there).
2. A correlation of 0.8 can indicate just as good alignment of corresponding structures as a correlation of 0.9. How can this be? Well, we have two parameters that change between cycles: (1) the physical location of the sample; (2) the intensity of the signal due to bleaching. Registration is maximizing the correlation by "moving" one image with respect to the other. When the signal intensity changes due to bleaching we will not get a correlation value of 1 even when the two images are perfectly aligned. What is more likely is that the maximum correlation will still correspond to the correct transformation/alignment but with a lower correlation value (e.g. 0.8).

Finally, out of curiosity, what were the correlation values prior to registration?

**Comment by Katitamorais** [comment]
Hello **### @zivy** 

I added some images after aligment. In some samples we have (PC133 and PC265 in the pdf file) alignment discrepancies and, 
coincidentally, low values ​​of correlation between cycles. These discrepancies impact downstream analysis, mostly cells segmentation mask.

We will try elastix-napari plugin.

Thank you for all explanation and support.

[Image.Registration.Simple.ITK.pdf](https://github.com/user-attachments/files/16394245/Image.Registration.Simple.ITK.pdf)

**Comment by zivy** [comment]
Hello @Katitamorais,

Based on the additional information from the pdf you provided it appears that the correlation, prior to registration, in the failed cases is close to zero. This is a bit suspicious, I've never seen it be so uncorrelated. Can you share the input imaris images for one of these cases so that I can take a deeper look? If yes, please use google drive, box or a similar service and share the download link here.

**Comment by Katitamorais** [comment]
Hello @zivy, 

I shared all files! I sent you a link by email.

Thank you.

**Comment by zivy** [comment]
Hello @Katitamorais,

I tried accessing the data via the link provided in your email. I get a "This folder is empty" message, please confirm you uploaded the data to the google drive location you shared.

**Comment by Katitamorais** [comment]
Hi @zivy,

sorry,  I don't know what happened. When I check it, all files are in the Google Drive.  I will reupload.

<img width="964" alt="Captura de Tela 2024-07-29 às 16 26 29" src="https://github.com/user-attachments/assets/fe5e3b1e-16b4-4729-b6de-36bbb07fe459">

**Comment by zivy** [comment]
Hi @Katitamorais,

Thanks for sharing the data. I registered it using the [XTRegisterSameChannel](https://niaid.github.io/imaris_extensions/XTRegisterSameChannel.html), affine registration program. When looking at sample PC133 I see that it has a central region that is common to all samples but the periphery changes considerably. I am not sure if these are expected sample changes over time or if it is a problem with the data acquisition (unexpected deformations).

[This pdf](https://github.com/user-attachments/files/16431724/PC133_registration_input_output.pdf) shows the input images from the three panels and the registration result. In the combined image after registration, the DAPI channel for panel1 is in blue, panel2 red and panel3 green. When all three overlap we get white. This highlights good overlap obtained by the registration in the central region and problems in the periphery.

Before recommending that you try deformable registration I'd like to ask folks in the community to chime in with respect to the issues encountered in this dataset so that if you decide to go this route you have a good rational for it.

Why the caution? With deformable registration you may force overlaps that are physically incorrect but look visually plausible. Registration is optimizing image similarity and if we use a transformation with more degrees of freedom than the actual physical transformation (deformable when in the physical world it was affine) we will end up with visually pleasing results that are incorrect. Think of all those videos where you see one object morphed into another. We don't want to morph an orange into an apple as shown in this [YouTube video](https://www.youtube.com/watch?v=M2OnhrkWGbg).

**Reply by radtkea** [reply]
Hi @Katitamorais and @zivy,

Thank you both for asking and answering this question in the open! I am sure it will help others with a similar problem.

![Screenshot 2024-08-01 233415](https://github.com/user-attachments/assets/5a9e4047-5b3e-49e9-8596-7daf112e6a08)

Based on the image overlay, it looks like the center part of the tissue is aligned but the outer perimeter is not. We sometimes see tissue lifting on the very outer edges of the sample but not to this extent. 

Some questions for you:
1. What kind of tissue is this, e.g., fixed frozen human jejunum or FFPE human lung?
2. What is the thickness of the tissue section, e.g., 10 um or 5 um?
3. Important: What is the expiration date of your chrome alum gelatin? Tissue adherence is impacted by the age of the adhesive. I always replace it 3 months before it expires.
4. Is there something unique about the outer edge of the tissue? Is it particularly fatty or mostly connective tissue?

Thanks very much,
A

**Comment by Katitamorais** [comment]
Hello!

@zivy and @radtkea, sorry for my delay! I spent my time on the bench and in an external scientific meeting during the last days.

@zivy, thank you very much for verifying our images. I would like to test deformable registration and see the result. I thought that running an analysis (cell segmentation, cell phenotype quantification) only in the good overlap region (white central region for PC133) could be an option to not lose this patient's sample, but I am not sure.

@radtkea, here are the answers:
1) skin lesions tissues form cutaneous leishmaniasis patients - FFPE
2) 5 um
3) when slides were  mounted and acquired, the expiration date was ok
4) connective tissue

This situation illustrates well the importance of adjustments during confocal acquisition, as well as care to avoid tissue damage. 

Thanks.

**Reply by radtkea** [reply]
Thank you @Katitamorais for providing details about your sample. It looks like everything should work and you shouldn't see too much tissue lifting because you are taking care to avoid tissue damage, using thin FFPE tissue sections which tend to adhere to the slide well, and using chrome gelatin alum within the expiration date. However, skin can be a bit delicate.

How did everything go with the deformable registration? Let us know if you have any further questions!

Best of luck with your bench work and I hope you had a great time at your meeting!
